### PR TITLE
Flip the Data Explorer

### PR DESCRIPTION
**Pull request by alexpaxton**
### Problem
All the important UI is at the bottom, graph is over-emphasized

### Solution
Flip the halves of the DE such that the query builder is on top. Draws a lot more attention to the query box, and editor as a whole

screen shot 2017-04-06 at 1 03 49 pm

### CODE REVIEW

**Comment by alexpaxton** [comment]
Uncovered some issues after making the flip:

- Dygraphs Legend needs special positioning rules in the DE
- Resizing the DE does not resize the graph until window-resize (or next refresh interval) is triggered

**Comment by lukevmorris** [comment]
Heya @alexpaxton. Those console errors look related to the issues you're seeing with the graph not resizing. We're wiring a `width` and `height` value from the top level component (`QueryBuilder`) all the way down to the graph (`Dygraph`). If `Dygraph` doesn't see the height value change, it doesn't know to rerender. Since it was resizing fine in the original configuration, it looks like the refactor has broken that wire somewhere.

**Comment by cryptoquick** [comment]
I have added a fix for @alexpaxton 's issue.

**Comment by alexpaxton** [comment]
@cryptoquick yooo I get this error after using the resizer:

```
Warning: Failed prop type: Invalid prop `height` of type `number` supplied to `Visualization`, expected `string`.
    in Visualization (created by DataExplorer)
    in ResizeBottom (created by DataExplorer)
    in div (created by ResizeContainer)
    in ResizeContainer (created by DataExplorer)
    in div (created by DataExplorer)
    in DataExplorer (created by Connect(DataExplorer))
    in Connect(DataExplorer) (created by App)
    in div (created by App)
    in App (created by withRouter(App))
    in withRouter(App) (created by RouterContext)
    in CheckSources (created by withRouter(CheckSources))
    in withRouter(CheckSources) (created by Connect(withRouter(CheckSources)))
    in Connect(withRouter(CheckSources)) (created by RouterContext)
    in div (created by App)
    in App (created by Connect(App))
    in Connect(App) (created by RouterContext)
    in RouterContext (created by Router)
    in Router (created by Root)
    in Provider (created by Root)
    in Root
```

**Comment by cryptoquick** [comment]
Easily corrected. One meowment. 😸

**Comment by cryptoquick** [comment]
@alexpaxton that should do the trick.

File: ui/src/data_explorer/components/FieldList.js [hunk 1]
```diff
@@ -98,11 +98,11 @@ const FieldList = React.createClass({
 
     return (
       <ul className="qeditor--list">
-        {this.state.fields.map((fieldFunc) => {
+        {this.state.fields.map((fieldFunc, i) => {
           const selectedField = this.props.query.fields.find((f) => f.field === fieldFunc.field)
           return (
             <FieldListItem
-              key={fieldFunc.field}
+              key={fieldFunc.field + i}
```
Comment (121watts) [inline ui/src/data_explorer/components/FieldList.js]: I'm not sure why this is necessary.  Fields should be unique per db.

File: ui/src/data_explorer/components/Visualization.js [hunk 2]
```diff
@@ -112,7 +112,7 @@ const Visualization = React.createClass({
 
   renderTable(query, heightPixels, onEditRawStatus) {
     if (!query) {
-      return <div className="generic-empty-state">Enter your query below</div>
+      return <div className="generic-empty-state">Enter your query above</div>
```
Comment (121watts) [inline ui/src/data_explorer/components/Visualization.js]: Probably a good idea. hehe.

File: ui/src/shared/components/Dygraph.js [hunk 3]
```diff
@@ -122,7 +124,12 @@ export default React.createClass({
         }
 
         legendContainerNode.style.left = `${legendLeft}px`
-        legendContainerNode.style.top = `${legendTop}px`
+        if (legendOnBottom) {
+          legendContainerNode.style.bottom = `4px`
+        } else {
+          legendContainerNode.style.top = `${legendTop}px`
+        }
```
Comment (121watts) [inline ui/src/shared/components/Dygraph.js]: Not sure of the ramifications of this on the rest of the app.

**Comment by lukevmorris** [comment]
Reviewing now. I'll pull the code down and play with it

File: ui/src/shared/components/LineGraph.js [hunk 4]
```diff
@@ -119,6 +119,7 @@ export default React.createClass({
           dygraphSeries={dygraphSeries}
           ranges={ranges || this.getRanges()}
           ruleValues={ruleValues}
+          legendOnBottom={isInDataExplorer ? true : null}
```
Comment (121watts) [inline ui/src/shared/components/LineGraph.js]: Why dont you just use the isInDataExplorer prop?  The ternary looks unnecessary

File: ui/src/style/pages/data-explorer/font-scale.scss [hunk 5]
```diff
@@ -26,6 +26,9 @@ $breakpoint-c: 2100px;
     .query-builder--tabs {
       width: 320px;
     }
+    .query-builder--tab-label {
+      width: (320px - 8px - 16px - 16px);
```
Comment (lukevmorris) [inline ui/src/style/pages/data-explorer/font-scale.scss]: Does this need to be a calc?

File: ui/src/style/pages/data-explorer/font-scale.scss [hunk 6]
```diff
@@ -89,6 +92,7 @@ $breakpoint-c: 2100px;
       letter-spacing: 0.3px;
     }
     .query-builder--tab-label {
+      width: (373px - 8px - 16px - 16px);
```
Comment (lukevmorris) [inline ui/src/style/pages/data-explorer/font-scale.scss]: And this?

File: ui/src/style/pages/data-explorer/query-builder.scss [hunk 7]
```diff
@@ -132,7 +135,7 @@
   font-weight: 600;
   white-space: nowrap;
   overflow: hidden;
-  width: 90%;
+  width: ($query-builder-tabs-width - 8px - 16px - 16px);
```
Comment (lukevmorris) [inline ui/src/style/pages/data-explorer/query-builder.scss]: Is this missing a calc?

**Pull request merged**